code/game/gamemodes/clock_cult/clock_structures/ark_of_the_clockwork_justicar.dm

```diff
@@ -14,18 +14,34 @@
 	can_be_repaired = FALSE
 	immune_to_servant_attacks = TRUE
 	var/progress_in_seconds = 0 //Once this reaches GATEWAY_RATVAR_ARRIVAL, it's game over
+	var/wisdom_regen_cycle = 0 //How many ticks It's been between wisdom regenerations
 	var/purpose_fulfilled = FALSE
 	var/first_sound_played = FALSE
 	var/second_sound_played = FALSE
 	var/third_sound_played = FALSE
 	var/fourth_sound_played = FALSE
 	var/obj/effect/clockwork/overlay/gateway_glow/glow
 	var/obj/effect/countdown/clockworkgate/countdown
-	var/list/required_components = list(BELLIGERENT_EYE = ARK_CONSUME_COST, VANGUARD_COGWHEEL = ARK_CONSUME_COST, GEIS_CAPACITOR = ARK_CONSUME_COST, REPLICANT_ALLOY = ARK_CONSUME_COST, HIEROPHANT_ANSIBLE = ARK_CONSUME_COST)
 
 /obj/structure/destructible/clockwork/massive/celestial_gateway/Initialize()
 	. = ..()
-	INVOKE_ASYNC(src, .proc/spawn_animation)
+	START_PROCESSING(SSprocessing, src)
+	if(GLOB.ark_of_the_clockwork_justiciar)
+		qdel(src)
+		return
+	else
+		GLOB.ark_of_the_clockwork_justiciar = src
+	var/turf/T = get_turf(src)
+	var/list/open_turfs = list()
+	for(var/turf/open/OT in orange(1, T))
+		if(!is_blocked_turf(OT, TRUE))
+			open_turfs |= OT
```

Comment:
You should not be using ```|=``` here, the purpose of ```|=``` is to prevent duplicate entries.
It is slow.
It is also not needed here as the list is local to this proc, and is not added to after being set up, and a turf cannot appear twice in 1 ```orange()``` call.